### ISSUE TITLE
[l10n] "Trackers blocked" string truncates in multiple locales

### ISSUE THREAD

**Issue by Delphine** [mozilla-mobile/focus-android#2341]
It seems like this fell through the cracks somehow. The "Trackers blocked" string in the menu ends up truncating in many locales. While some are able to shorten the string, many can not find a work around this.
See for example Italian (has since been shortened):  https://npark-mozilla.github.io/it/images/phoneScreenshots/BrowserViewMenu_1521227033320.png

You can see which locales are affected here: https://npark-mozilla.github.io/index.html
It seems at least 28 locales are affected: ast, bg, bn-BD, bn-IN, cs, da, dsb, el, es-CL, es-ES; es-MX, eu, hsb, hy-AM, ja, ka, kk, lt, mix, my, pt-BR, ru, sk, trs, tt, uk, vi and wo.

While many of these are our "smaller" locales, as you can see some in there are major ones. And as I mentioned, while some were able to fix this by shortening the strings, it's far from ideal as some just can't.

Also, the main issue is that the word that ends up being truncated is "blocked" in most cases, which means the sense of the string is entirely lost.

**Comment by pocmo** [comment]
@brampitoyo What are your thoughts?

**Comment by brampitoyo** [comment]
es, ja, pt-BR and ru are big locales! @Delphine Would you be able to tell me which locale has the longest string that wouldn’t fit, and the content of the string?

There are at least two things we can try here:

1. Allow the menu width to expand as wide as the content wants it, with a maximum width of screen width minus 8dp (to add a good margin).
2. Reduce the text size of “Trackers blocked”, down to 14dp.

After getting the string from @Delphine, I’ll try these approaches and post my result to this issue.

**Comment by Delphine** [comment]
@brampitoyo: Thanks for looking into this, very helpful! 
So as a general rule, we usually say to estimate that translations will end up being at least 30% longer than an English string. That helps estimate if a string will fit once localized or not.

Here it's hard to say which string is the longest since I don't speak all languages / don't know all the fonts, but from a quick look at Transvision: https://transvision.mozfr.org/string/?entity=focus_android/app.po:0d6ef5cb439f059884232d5680ac138f&repo=focus_android
it seems that the longest would be Slovak (sk) in this case "Počet zablokovaných sledovacích prvkov" with a grand total of 38 letters.

**Comment by pocmo** [comment]
> Allow the menu width to expand as wide as the content wants it

This could look awkward and is different than any other menus on Android.

> Reduce the text size of “Trackers blocked”, down to 14dp.

Let's see if this fixes the issue for Slovak. :)

Are there any other options how we could re-design this block so that we either end up with shorter or different strings?

**Comment by brampitoyo** [comment]
Right. Other than reducing the size down to 14dp, which probably won’t fix the issue for Slovak, what about moving the overflow text to a new line? Yes, the Tracking Protection area may be taller on some locales as a result, but it will always fit the text.

In terms of engineering, will it be possible to make the height of this area dynamic, depending on text length?

**Comment by pocmo** [comment]
> In terms of engineering, will it be possible to make the height of this area dynamic, depending on text length?

Yes! Currently it is set to be only one line explicitly.

**Comment by brampitoyo** [comment]
@pocmo Perfect! Extending the “Trackers blocked” text so it occupies two lines on certain locales sounds like a good solution. I suspect that’s all we need for even the longest locale.

We can even potentially extend it to three if necessary – this is far from ideal, though.

So, to summarise, we should try this solution in order, and stop if doing just one or two will fix the issue for the longest locale:

1. Reduce text size
2. Make text occupy two lines
3. Make text occupy three lines

**Comment by pocmo** [comment]
> Make text occupy two lines
> Make text occupy three lines

We can also set it to no specific number of lines and let it grow to whatever it needs. Or do you explicitly want the constraint?

**Comment by brampitoyo** [comment]
@pocmo No need for constraint. Just let it grow to as tall as it needs to be. I originally thought (mistakenly) that maximum value cannot be removed.

**Comment by brampitoyo** [comment]
The UX work for this issue is now finished.
